coordinator/src/context/chain_history_access.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::header::Header;
```

Comment:
Ah, I forgot to mention it.
Thank you for pointing it.
I fixed it because it looks weird to me that we already added a new simplified type, but it returns an original header type. The differences between the two types are the parent root, evidences_root, transactions_root, state_root, next_validator_set_hash, and seal. Of course, the hash of the block cannot be calculated with the simplified header.
@somniumism How about you? Do you think we need theses information in the module?

Reply:
Oh, I couldn't check this change because the change was invisible to a383e03 . I had misunderstood the header type of the coordinator and host side, and it was just resolved through discussion with Byeongjee. This change looks good to me.

Reply:
@Byeongjee @somniumism While I'm porting the staking module, I meet the case that the module needs more field than the currently provided fields. The staking module needs the parent block hash field. I think there are two solutions. The first is to add the parent hash to the simplified header, the other is to make the ChainHistory return the full header.
I think the both options are reasonable. How about you?

Reply:
This PR defines `ChainHistoryAccess` that can return the full header type from the host side. Could you explain the difference between `ChainHistoryAccess` and the latter you said?

Reply:
@Byeongjee What do you think? : )

Reply:
@somniumism This PR defines `ChainHsitoryAccess` to return the simplified header. The `crate::header::Header` is simplified one. The `Header` defined in `ctypes` is the full header. The simplified one has only some fields that are used by the coordinator. The original implementation in the mold branch returns the full header, but I changed it to return the simplified one because I'm not sure whether the module needs to access all fields in the header.
I met the case that needs more fields, but I'm still wondering whether the module needs to access all header fields, and is it a good design to provide the full header. Because the most bytes of the header would be the signatures. They are merely the waste of the traffics in the most usage.

Reply:
@somniumism @sgkim126 I think it's better to put some additional fields to the simplified header and use that.
Note that we should put some simplified vote info to our simplified header. @HoOngEe did it.

Reply:
Okay. I'll merge this commit and add fields that the modules need.